### PR TITLE
chore(router) deprecate setting path handling v1 when router_flavor is traditional_compatible

### PR DESCRIPTION
**Pull request by windmgc**
### Summary

This PR ~~denies~~ users from setting `path_handling=v1` when the Kong gateway is running with `router_flavor=traditional_compatible`, as the `path_handling` feature is stepping into deprecation and path_handling v0 will be the default behaviour when composing upstream_uri.

Discussed with the team and decide to only print deprecation log in the current version.

### Full changelog

* Add custom entity check function to deny setting `path_handling=v1` when `router_flavor` is `traditional_compatible`

### CODE REVIEW

**Comment by chronolaw** [comment]
Good Job. 

But I think both `traditional_compatible` and `expression` don't support `v1`, so we should change the condition to `kong.configuration.router_flavor ~= "traditional"`

**Comment by windmgc** [comment]
@chronolaw Do you mean `router_flavor = atc`?

I saw that router_flavor only have three enum results
https://github.com/Kong/kong/blob/9c990a79083f1b529eac69238552ec55f59564ed/kong/conf_loader/init.lua#L438-L440

And when `router_flavor = atc` the schema will be a different table without the `path_handling` 
https://github.com/Kong/kong/blob/9c990a79083f1b529eac69238552ec55f59564ed/kong/db/schema/entities/routes.lua#L6-L42

**Comment by chronolaw** [comment]
Now `atc` will rename to `expression`, see #9263 .

File: kong/db/schema/entities/routes.lua [hunk 1]
```diff
@@ -122,6 +122,15 @@ else
                         then_match = { len_eq = 0 },
                         then_err = "'snis' can only be set when 'protocols' is 'grpcs', 'https', 'tls' or 'tls_passthrough'",
                       }},
+      { custom_entity_check = {
+        field_sources = { "path_handling" },
+        fn = function(entity)
+          if entity.path_handling == "v1" and kong.configuration.router_flavor ~= "traditional" then
+            return nil, "path_handling = 'v1' is deprecated and not supported with router_flavor = 'traditional_compatible'"
```
Comment (chronolaw) [inline kong/db/schema/entities/routes.lua:129]: We should also change this error message.

**Comment by chronolaw** [comment]
I think it is not very urgent for 3.0, because the new router only supports `v0`.

**Comment by windmgc** [comment]
Yes, agree, not a blocker for 3.0, so no hurry for merging.
And I still need to get the tests fixed :P

File: kong/db/schema/entities/routes.lua (hunk 1 above)
Comment (windmgc) [reply kong/db/schema/entities/routes.lua]: done

File: kong/db/schema/entities/routes.lua [hunk 2]
```diff
@@ -126,7 +126,7 @@ else
         field_sources = { "path_handling" },
         fn = function(entity)
           if entity.path_handling == "v1" and kong.configuration.router_flavor ~= "traditional" then
-            return nil, "path_handling = 'v1' is deprecated and not supported with router_flavor = 'traditional_compatible'"
+            return nil, "path_handling = 'v1' is deprecated and not supported when router_flavor is not 'traditional'"
```
Comment (chronolaw) [inline kong/db/schema/entities/routes.lua:129]: How about `only supported when router_flavor is 'traditional'`

Comment (windmgc) [reply kong/db/schema/entities/routes.lua]: done

**Review by chronolaw — APPROVED** [review]

File: kong/db/schema/entities/routes.lua [hunk 3]
```diff
@@ -122,6 +125,23 @@ else
                         then_match = { len_eq = 0 },
                         then_err = "'snis' can only be set when 'protocols' is 'grpcs', 'https', 'tls' or 'tls_passthrough'",
                       }},
-                    },
+      { custom_entity_check = {
+        field_sources = { "path_handling" },
+        fn = function(entity)
+          if entity.path_handling == "v1" then
+            if kong_router_flavor == "traditional" then
+              deprecation("path_handling='v1' is deprecated and will be removed in future version, " ..
```
Comment (mayocream) [inline kong/db/schema/entities/routes.lua:133]: Shall we add a deprecation after version and a removal version?
https://github.com/Kong/kong/blob/e2b048845429694f009b73d3a482cd0182af494c/kong/conf_loader/init.lua#L681-L685

File: kong/db/schema/entities/routes.lua [hunk 4]
```diff
@@ -122,6 +125,23 @@ else
...
+                          "please use path_handling='v0' instead")
+
+            elseif kong_router_flavor == "traditional_compatible" then
+              deprecation("path_handling='v1' is deprecated and will not work under traditional_compatible " ..
```
Comment (mayocream) [inline kong/db/schema/entities/routes.lua:137]: ditto

**Review by mayocream — APPROVED** [review]

**Pull request merged**